libr/core/cmd_type.c

```diff
@@ -745,6 +746,40 @@ static int cmd_type(void *data, const char *input) {
 				r_core_cmd_help (core, help_msg_ts);
 			}
 			break;
+		case 'c': { //tsc
+			char *name = NULL;
+			char *varname = NULL;
+           		SdbKv *kv;
+           		SdbListIter *iter;
+			int i, j;
+        		SdbList *l = sdb_foreach_list (TDB, true);
+        		ls_foreach (l, iter, kv) {
+        			if (!strcmp (sdbkv_value (kv), "struct")) {
+                			if (!name || strcmp (sdbkv_value (kv), name)) {
+                				free (name);
+                    				name = strdup (sdbkv_key (kv));
```

Comment:
Probably starting from here?

Reply:
i dont see why you need to hold a copy of the key name in the heap.